lib/analyze_config_file_diffs_task.rb

```diff
@@ -23,6 +23,18 @@ def analyze(description)
       "repositories",
       "config_files"
     )
+    online_repo = 0
+    description["repositories"].each do |repo|
+      if !repo["url"].start_with?("dvd:///") &&
+        !repo["url"].start_with?("cd:///") &&
+        repo["enabled"] == true
+        online_repo =+ 1
+      end
+    end
+    if online_repo == 0
+      raise Machinery::Errors::UpgradeFailed.new("No online repositories available.")
```

Comment:
I'd maybe use a friendlier message, e.g. `Can not analyze the system description because it does not contain any online repository` or so.